InField Quickstart: Example on how to set up modules

erikangelsen-cognite opened the pull request:
Description
This PR has an example on how to configure one location for InField Quickstart

- LocationFilter
- InFieldLocationConfig
- Group
- Spaces

erikangelsen-cognite committed:
InField Quickstart: Example on how to set up modules

spex66 commented on modules/infield_quickstart/cdf_infield_location/auth/infield_quickstart_normal_role.group.yaml:
```diff
@@ -0,0 +1,69 @@
+        - READ
+        - LIST
+      scope:
+        all: {}
```
That should be scoped to `currentuserscope: {}`, only root-accounts (admin to run cdf-tk for example) should get all-scope

spex66 commented on modules/infield_quickstart/cdf_infield_location/README.md:
```diff
@@ -0,0 +1,62 @@
+
+## Auth
+
+The module creates one group that needs a matching group in the identity provider that the CDF project is configured with:
```
I've limited knowledge about Infield but from my knowledge of InRobots you might need an additional 'Infield Admin' group, which can write to / change `APM_config`?

spex66 commented on modules/infield_quickstart/cdf_infield_location/auth/infield_quickstart_normal_role.group.yaml:
```diff
@@ -0,0 +1,69 @@
+      scope:
+        idScope:
+          ids:
+            - ext_infield_{{first_location}}_config # this is an internal id, how can we set this? lets try with externalId
```
Why is it prefixed `ext_`?

erikangelsen-cognite replied:
just to show its an externalId. I saw that everything else was `gp` -> group, `sp` -> space etc.
Not important to do so, but found it more readable.

erikangelsen-cognite committed:
Update infield quickstart configuration and auth group

- Add comment explaining first_location variable in default.config.yaml
- Fix auth group space references

erikangelsen-cognite committed:
set up infield quickstart as selectable template

erikangelsen-cognite committed:
fix package id

erikangelsen-cognite committed:
add packages.zip

erikangelsen-cognite committed:
add InFieldOnCDM data model to locationsFilter

erikangelsen-cognite committed:
Clean up infield quickstart configuration

- Remove unused default.config.yaml from cdf_infield_common module
- Remove commented-out file ACL section from auth group configuration

erikangelsen-cognite commented:
should we set up a new docs page for quickstart like this?: https://github.com/cognitedata/docs.cognite.com/blob/054f7de7a15304065242172824fa5aa11bb09b34/docs/cdf/deploy/cdf_toolkit/guides/set_up_infield.mdx#L4

ronpal reviewed (changes requested):
Can you please remove the packages.zip file? We want the CI/CD process to build it :)

erikangelsen-cognite committed:
remove packages.zip
